### ISSUE TITLE
Warning that `self` is undocumented

### ISSUE THREAD

**Issue by rdong8** [doxygen/doxygen#11123]
**Describe the bug**
Describe what you see that (you think) is wrong.

The C++23 explicit object parameter ([standard reference](https://eel.is/c++draft/dcl.fct#6)) needs to be documented to avoid getting a warning for an undocumented parameter by Doxygen. It would be nice if there was a flag to turn only this off without disabling warnings for any other undocumented component.

**Screenshots**
If useful, add screenshots to help explain your problem.

**To Reproduce**
Attach a self contained example that allows us to reproduce the problem.
Such an example typically exist of some source code (can be dummy code) and a doxygen configuration file used (you can strip it using `doxygen -s -u`). After you verified the example demonstrates the problem, put it in a zip (or tarball) and attach it to the bug report. Try to avoid linking to external sources, since they might disappear in the future.

[example.zip](https://github.com/user-attachments/files/16921736/example.zip)

Run:

```bash
make config build
```

This appears in the output:

```
/home/rd8/tmp/doxy/main.cc:8: warning: The following parameter of Foo::f(this auto const &self, int x) -> void is not documented:
  parameter 'self'
```

**Expected behavior**
Describe what you would have expected or think is correct.

Doxygen should not emit a warning that `self` is undocumented.

**Version**
Mention the version of doxygen used (output of `doxygen --version`) and the platform on which you run doxygen (e.g. Windows 10, 64 bit). If you run doxygen under Linux please also mention the name and version of the distribution used (output of `lsb_release -a`) and mention if you compiled doxygen yourself or that you use a binary that comes with the distribution or from the doxygen website.

```
1.10.0 (GIT-NOTFOUND)
```

**Stack trace**
If you encounter a crash and can build doxygen from sources yourself with debug info (`-DCMAKE_BUILD_TYPE=Debug`), a stack trace can be very helpful (especially if it is not possible to capture the problem in a small example that can be shared).

**Additional context**

**Comment by albert-github** [comment]
The example given is a bit incomplete to reproduce the issue:
- `;` missing at the end of the class
- is the function `public`?
- which settings are being used?

With some default settings (and adding the `;` and making `f` public` I don't get a warning about that `self` is not documented (as nothing is documented at all) but just:
```
aa.h:1: warning: Compound Foo is not documented.
aa.h:4: warning: Member f(this auto const &self) -> void (function) of class Foo is not documented.
```

- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).
- A reference to the C++ 2023 standard would also be good as that is normative. (Looks like in https://en.wikipedia.org/wiki/C%2B%2B23 it is called "explicit this object parameter". Looks like it might be described in the standard under "9.3.4.6 Functions" and that the final proposal is in https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p0847r7.html

**Comment by rdong8** [comment]
Apologies for the errors, the post has been updated.

**Comment by albert-github** [comment]
Would have been nice to have a `Doxyfile` with the question instead of an (outdated) file with settings in `CMakeLists.txt` (`CLASS_DIAGRAMS`, `CLASS_GRAPH` and `DOT_TRANSPARENT` have been updated / are obsolete and anyway not relevant for this issue)

Doxyfile: [Doxyfile.tar.gz](https://github.com/user-attachments/files/16921718/Doxyfile.tar.gz)

**Comment by rdong8** [comment]
Ah yes sorry, copied from another project, just didn't update the zip after I removed them. Have done so now + the equivalent Doxyfile.

**Comment by doxygen** [comment]
@rdong8 Please verify if the referenced commit fixes the problem for you. Do not close the issue, this will be done automatically when the next official release becomes available.

**Comment by rdong8** [comment]
I get this error when building from source with a Fedora 41 container with clang 19.1.0:

```
doxygen: error while loading shared libraries: libiconv.so.2: cannot open shared object file: No such file or directory
```

Note that I've built libiconv like this:

```bash
wget https://ftp.gnu.org/pub/gnu/libiconv/libiconv-1.17.tar.gz
gunzip libiconv-1.17.tar.gz
tar -xf libiconv-1.17.tar
cd libiconv-1.17
./configure --prefix=/usr/local
make
make install
```

Note that `ls /usr/local/lib` results in:

```
libcharset.a  libcharset.la  libcharset.so  libcharset.so.1  libcharset.so.1.0.0  libiconv.la  libiconv.so  libiconv.so.2  libiconv.so.2.6.1                              │
```

Then I've built doxygen like this:

```bash
git clone https://github.com/doxygen/doxygen.git
mkdir doxygen/build
cd doxygen/build
cmake -G "Unix Makefiles" ..
make
make install
```

**Comment by albert-github** [comment]
@rdong8 

Looks a bit like a problem with the library path, is `/usr/local/lib` in your `LD_LIBRARY_PATH`?

Not sure whether or not an Ubuntu executable will run on your system, but maybe have a look at the artifacts (botom of the page) at https://github.com/doxygen/doxygen/actions/runs/11645540254

**Comment by rdong8** [comment]
Yep thanks that was it. Issue seems to be fixed.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
